build-system/tasks/performance/index.js

```diff
@@ -14,33 +14,75 @@
  * limitations under the License.
  */
 
-const cacheDocuments = require('./cache-documents');
-const compileScripts = require('./compile-scripts');
-const getMetrics = require('./measure-documents');
+const argv = require('minimist')(process.argv.slice(2));
 const loadConfig = require('./load-config');
-const rewriteScriptTags = require('./rewrite-script-tags');
 const runTests = require('./run-tests');
-const {installPackages} = require('../../common/utils');
+const {
+  buildMinifiedRuntime,
+  buildUnminifiedRuntime,
+  installPackages,
+} = require('../../common/utils');
+const {CONTROL, EXPERIMENT, HOST, PORT} = require('./helpers');
+const {measureDocuments, writeMetrics} = require('./measure-documents');
 const {printReport} = require('./print-report');
+const {startServer, stopServer} = require('../serve');
+
+async function setupExperimentTests() {
+  installPackages(__dirname);
+  if (!argv.nobuild) {
+    if (argv.compiled) {
+      buildMinifiedRuntime();
+    } else {
+      buildUnminifiedRuntime();
+    }
+  }
+
+  await startServer(
```

Comment:
We still want to modify the inline analytics config to add some special urlParams to analytics outgoing urls (so we can easily detect them). 
What's your recommendation on supporting that?

Reply:
Since we're testing manually created pages, we could manually add the special urlParams to each analytics config. Additionally, we could use puppeteer to intercept the outgoing request for the page itself, and rewrite the analytics config in the response. Finally, (not sure if feasible), we could rewrite the config before we serve it. Thoughts @zhouyx?

Reply:
> Additionally, we could use puppeteer to intercept the outgoing request for the page itself

My question is: are we still able to do this with all files served from the local server, or we need to add proxy to testing server instead?

Reply:
Discussed offline: puppeteer doesn't support response interception, so a good approach instead would be to check all the test urls (within our local directory), rewrite their configs if necessary, and then start up the server. (Thanks @estherkim)